### ISSUE TITLE
Consider to change the return value when Eloquent model is updated

### ISSUE THREAD

**Issue by Vitaliy-1** [pkp/pkp-lib#10487]
**Describe the bug**
Originally, when eloquent model is updated, the method returns the number of updated rows from the primary table. E.g.,
```php
Announcement::where('type_id', '=', '1')->update([
  'title' => 'Announcement Title'
  'type_id' => '2'
])
```
After refactoring, Models that are utilising settings table, will return the number of updated rows from the primary table and settings table.

1. Explore if this has any drawbacks.
2. Consider alternative option, return associative array with the number of updated rows from primary and settings table (https://github.com/pkp/pkp-lib/pull/10382#discussion_r1771402157) 

**What application are you using?**
 OJS main branch

**Comment by touhidurabir** [comment]
I think we should try to preserve the default behaviours of Eloquent and Builders as much as possible and if need to changed, should be changed for all cases . Having the `update` return  2 different types of value structure (int and array) only given that when there is settings table associated will make it harder to track how to use the return values .

**Comment by touhidurabir** [comment]
@Vitaliy-1 with the merge of https://github.com/pkp/pkp-lib/issues/10562 , now we are only considering on the `primary table` updated rows count only and not the `settings table` rows count anymore . I think this is fine and thats how it should be . If we really need any settings table rows count, then probably it's better to have different mechanism . 

So considering the above case, should we close this issue as `not planned` ?

**Comment by asmecher** [comment]
Agreed with @touhidurabir, we should stick with the Eloquent upgrade function's return behaviour. If we really need to know the number of settings rows affected, then it should use a mechanism other than the return value (e.g. maybe an optional by-reference parameter, if that doesn't conflict with the standard Laravel parameter list overmuch).

@Vitaliy-1 and @Hafsa-Naeem, can you describe the use case where we need both counts? It looks like it arose from [this issue/comment](https://github.com/pkp/pkp-lib/pull/10382#discussion_r1760966869).
